tests/infra_test.go

```diff
@@ -984,3 +1037,19 @@ func validatedHTTPResponses(errorsChan chan error, concurrency int) error {
 
 	return sampledError
 }
+
+func getSupportedIP(virtClient kubecli.KubevirtClient, ips []string, family k8sv1.IPFamily) string {
+	ip := libnet.GetIp(ips, family)
+	if ip == "" {
+		if family == k8sv1.IPv4Protocol {
+			if tests.IsRunningOnKindInfraIPv6() {
+				Skip("Skip IPv4 on Single stack IPv6")
+			}
+		} else {
+			libnet.SkipWhenNotDualStackCluster(virtClient)
```

Comment:
It is enough to check whether the requested family is ipv6.
```
	if family == k8sv1.IPv6Protocol {
						libnet.SkipWhenNotDualStackCluster(virtClient)
					}
```
For better readability of the tests, I prefer to move the skip logic to the beginning of the `It`.
In `getSupportedIP` you should assume there is an ip. If `ip==""` you should fail.